### PR TITLE
multiline: respect comment and start space

### PR DESCRIPTION
**Pull request by changlong-liu**
fix issue #429 #404 #406 #428

### CODE REVIEW

File: src/test/scenarios/datafactory/output/src/datafactory/azext_datafactory/generated/_help.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
     examples:
       - name: Factories_ListByResourceGroup
         text: |-
-               az datafactory list --resource-group "exampleResourceGroup"
+               az datafactory list --resource-group "myResourceGroup2"
```
Comment (qiaozha) [inline src/test/scenarios/datafactory/output/src/datafactory/azext_datafactory/generated/_help.py:26]: Why this is myResourceGroup2

File: src/test/scenarios/datafactory/output/src/datafactory/azext_datafactory/generated/_help.py [hunk 2]
```diff
@@ -32,7 +32,7 @@
     examples:
       - name: Factories_Get
         text: |-
-               az datafactory show --name "exampleFactoryName" --resource-group "exampleResourceGroup"
+               az datafactory show --name "exampleFactoryName" --resource-group "myResourceGroup"
```
Comment (qiaozha) [inline src/test/scenarios/datafactory/output/src/datafactory/azext_datafactory/generated/_help.py:35]: I think what CLI team wanted is for all the resource name change it into myResourceName

Comment (qiaozha) [reply src/test/scenarios/datafactory/output/src/datafactory/azext_datafactory/generated/_help.py:35]: For ezample exampleFactoryName => myFactoryName

File: src/test/scenarios/datafactory/output/src/datafactory/azext_datafactory/generated/_help.py (hunk 1 above)
Comment (changlong-liu) [reply src/test/scenarios/datafactory/output/src/datafactory/azext_datafactory/generated/_help.py:26]: this should be the second resource-group in all example files

File: src/plugins/azgenerator/CodeModelAz.ts [hunk 3]
```diff
@@ -181,8 +181,10 @@ export interface CodeModelAz {
     GetPreparerEntities(): any[];
     GatherInternalResource();
     FindExampleWaitById(id: string): string[][];
+    GetExampleItems(example: CommandExample, isTest: boolean, commandParams: any): string[];
```
Comment (qiaozha) [inline src/plugins/azgenerator/CodeModelAz.ts:184]: I am wondering when do we call this GetExampleItems and pass isTest as true?

Comment (changlong-liu) [reply src/plugins/azgenerator/CodeModelAz.ts:184]: exists: https://github.com/Azure/autorest.az/blob/master/src/plugins/azgenerator/CodeModelAzImpl.ts#L1985

Comment (qiaozha) [reply src/plugins/azgenerator/CodeModelAz.ts:184]: Got it. thx

File: src/test/scenarios/datafactory/output/src/datafactory/azext_datafactory/generated/_help.py (hunk 1 above)
Comment (qiaozha) [reply src/test/scenarios/datafactory/output/src/datafactory/azext_datafactory/generated/_help.py:26]: Why? They are exampleResourceGroup in all the examples.

Comment (changlong-liu) [reply src/test/scenarios/datafactory/output/src/datafactory/azext_datafactory/generated/_help.py]: fixed here: https://github.com/Azure/autorest.az/pull/462/commits/6ec8b8a729c490e174ee53afdb633351604a1c0f

File: src/test/scenarios/datafactory/output/src/datafactory/azext_datafactory/generated/_help.py (hunk 2 above)
Comment (changlong-liu) [reply src/test/scenarios/datafactory/output/src/datafactory/azext_datafactory/generated/_help.py]: internal resource name are fixed in here: https://github.com/Azure/autorest.az/pull/462/commits/9bf198e7499790b8f59b7bae84850f9ec6167acd

**Pull request merged**